exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -44,6 +44,11 @@ type NodePoolMode string
 // AzureManagedMachinePoolSpec defines the desired state of AzureManagedMachinePool.
 type AzureManagedMachinePoolSpec struct {
 
+	// AdditionalTags AdditionalTags is an optional set of tags to add to Azure resources managed by the
+	// Azure provider, in addition to the ones added by default.
+	// +optional
+	AdditionalTags []string `json:"additionalTags,omitempty"`
```

Comment:
@zioproto you'll also need to use this field in the controller to make sure it's applied on the Azure resource. You can look at AzureMachine additionalTags for an example.